examples/columnar_clone.rs

```diff
@@ -0,0 +1,142 @@
+use core::any::TypeId;
```

Comment:
We should include module-level doc comments that explain what this demonstrates, why it's useful, why special effort is needed (there's no static guarantee that all components are `Clone`) and what the caveats are (unregistered components aren't cloned).

Reply:
Cool, I'll get on (I've not written an example before)